### PR TITLE
New group function for element grouping

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [x] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

The current `Group` class is made to group diagram items (`Presentation`). However, grouping allows on several places, like the tree view.

### What is the new behavior?

Grouping is now done on model element level. This allows us to use the same grouping logic in multiple places in the application.

`group()` and `ungroup()` is implemened in the dropzone tool, as well as the namespace model.
The namespace model does an extra check to see if the owner has actually changed. If that's not the case, the changes will be rolled back.

`group()` and `ungroup()` are idempotent: return `True` if a group can be set, or is already set. Return `False` otherwise.

All module names have changed from `-grouping` to `-group`.

### Does this PR introduce a breaking change?
- [x] Yes
- [ ] No

### Other information

This is a finding from the Hypothesis tests: adding grouping to the test model was not idiomatic.

Calling `rollback()` in a transaction context (`with` block) required a small change to the transaction code.

Fixed an issue in the UML model.

### CODE REVIEW

**Comment by danyeaw** [comment]
I've been slowly chipping away at reviewing this over the last few nights, hopefully I can finish it tomorrow :+1:

**Comment by amolenaar** [comment]
> I've been slowly chipping away at reviewing this over the last few nights, hopefully I can finish it tomorrow +1

I hope you like the changes. It took a while for me to understand what it means if grouping only works on model level.

**Review by danyeaw — APPROVED** [review]
This looks really great @amolenaar! This really adds a nice capability to bring up grouping to the model level :+1:

**Comment by amolenaar** [comment]
@danyeaw Thanks. Ne downside is that DnD in the tree view is now implicitly dependent on the undo manager.

I also think it makes a lot of stuff simpler 😀

File: gaphor/UML/actions/actionsgroup.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    """Group activity nodes within activity partition."""
+    if partition.subpartition:
+        return False
+    partition.node = node
+    return True
+
+
+@ungroup.register(ActivityPartition, ActivityNode)
+def activity_node_group(partition, node):
+    print("remove node")
```
[the remark this answers is not in the document]
Comment (amolenaar) [reply gaphor/UML/actions/actionsgroup.py]: That can go. Thanks for spotting :smiley:

**Pull request merged**